fe/fe-core/src/main/java/com/starrocks/catalog/ExprRangePartitionInfo.java

```diff
@@ -0,0 +1,184 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Range;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.CastExpr;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.util.RangeUtils;
+import com.starrocks.persist.gson.GsonUtils;
+import com.starrocks.server.RunMode;
+import com.starrocks.sql.analyzer.AnalyzerUtils;
+import com.starrocks.sql.analyzer.PartitionExprAnalyzer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * ExprRangePartitionInfo is an enhanced version of ExpressionRangePartitionInfo
+ * because ExpressionRangePartitionInfo is not easily scalable
+ * and get more extensions by extracting objects
+ * in the future this will replace all expr range partition info
+ */
+public class ExprRangePartitionInfo extends RangePartitionInfo {
+
+    private static final Logger LOG = LogManager.getLogger(ExprRangePartitionInfo.class);
+
+    @SerializedName(value = "exprPartitionMeta")
+    private ExprPartitionMeta exprPartitionMeta;
+
+    public ExprRangePartitionInfo() {
+        this.type = PartitionType.EXPR_RANGE_EX;
+    }
+
+    public ExprRangePartitionInfo(ExprPartitionMeta exprPartitionMeta, List<Column> columns) {
+        super(columns);
+        this.exprPartitionMeta = exprPartitionMeta;
+        this.type = PartitionType.EXPR_RANGE_EX;
+    }
+
+    public ExprPartitionMeta getExprPartitionMeta() {
+        return exprPartitionMeta;
+    }
+
+    public void setExprPartitionMeta(ExprPartitionMeta exprPartitionMeta) {
+        this.exprPartitionMeta = exprPartitionMeta;
+    }
+
+    public static PartitionInfo read(DataInput in) throws IOException {
+        ExprRangePartitionInfo info = new ExprRangePartitionInfo();
+        info.readFields(in);
+        String json = Text.readString(in);
+        ExprPartitionMeta meta = GsonUtils.GSON.fromJson(json, ExprPartitionMeta.class);
+        info.setExprPartitionMeta(meta);
+        List<Expr> exprs = meta.getPartitionExprs();
+        SlotRef slotRef;
+        for (Expr expr : exprs) {
```

Comment:
it seams check logic, move it somewhere else? eg: GsonPostProcessable?

Reply:
This is not to check the logic, but to complete the information corresponding to Expr after returning to serialization, or the serialization logic.